### ISSUE TITLE
Problem when converting and uploading in orthanc

### ISSUE THREAD

**Issue by LuisOtavioSantos** [imi-bigpicture/wsidicomizer#104]
Hello everyone,

Firstly I would like to thank you for this awesome project.
I'm using it to convert czi files to dicom, but the output dicom images are getting with this grid-lines between the tiles in orthanc wsi-viewer, can you help me with this? 

I've attempted to address this by using the '--offset-table' option with the value 'empty', but it hasn't yielded satisfactory results. 

`wsidicomizer --offset-table 'empty' --no-confidential -i '/home/user/files/czi_file.czi' -o '/home/user/files/dicom_output'   `

![Screenshot from 2024-04-22 13-03-56](https://github.com/imi-bigpicture/wsidicomizer/assets/45131350/5b153eeb-8ba4-4792-bd71-dd9017ae0146)

Additionally, I've tried setting the block size to 1024 with the command:

`wsidicomizer -t 1024 --offset-table 'empty' --no-confidential -i '/home/user/files/czi_file.czi' -o '/home/user/files/dicom_output'   `

![Screenshot from 2024-04-22 13-03-56](https://github.com/imi-bigpicture/wsidicomizer/assets/45131350/19ef4246-405c-4508-b32f-debf2cf9f579)

Thanks in advance

**Comment by erikogabrielsson** [comment]
I have found the bug in how the tile directory, specifying what czi blocks (overlapping images in the czi file) covers what tile. The bug resulted in some czi blocks not being recorded as covering a tile, resulting in white spaces.

Your workaround likely works because the change in tile size better matches the czi block organization.

I will make a bug-fix-release as soon as possible.

**Comment by erikogabrielsson** [comment]
@LuisOtavioSantos I have committed two fixes for czi files to [this branch](https://github.com/imi-bigpicture/wsidicomizer/tree/czi-fixes):

- Missing tiles in the block directory
- Deadlock on cache miss when reading block data.

You can test the branch with:
```
pip install git+https://github.com/imi-bigpicture/wsidicomizer.git@czi-fixes
```
Please let me know if this fixes the issue for you (using the larger tile size).

**Comment by LuisOtavioSantos** [comment]
Hi @erikogabrielsson!

Thank you for addressing the bug, the fixes you did improved my results. The file is taking less time for converting and opening in orthanc.

wsidicomizer --offset-table 'bot' -t 2048 --format 'jpeg' --quality 80 --subsampling '420' -i '/path/to/file.czi' -o '/path/to/dicom/output/folder'

![Screenshot from 2024-04-27 22-15-46](https://github.com/imi-bigpicture/wsidicomizer/assets/45131350/ad5a1005-43eb-48a5-bc89-928cd3963d28)

Thanks a lot!

**Comment by erikogabrielsson** [comment]
Good that it works for you to now.

Please not that the czi source only reads the base pyramid layer, and if you don't add `--add_missing_levels to your command you will only get the base level converted. I'm not sure how Orthanc handles DICOM WSIs without pyramid, but it is likely that it is better that you add the pyramid creation.

**Comment by erikogabrielsson** [comment]
Closed with #105
